### PR TITLE
[iOS] Fix unrecognized symbol `AVMetadataObjectTypeCodabarCode` in Xcode < 13.3 (without iOS 15.4)

### PR DESCRIPTION
**Pull request by bbarthec**
# Why

Our CI is failing on iOS jobs, because it runs on Xcode 13 and we have some codebase that is not available prior Xcode 13.3.
https://github.com/expo/expo/runs/5840230818?check_suite_focus=true
https://exponent-internal.slack.com/archives/CNHBN8LNS/p1649193243758459

# How

I've added macros guarding wrapping the code.

# Test Plan

iOS CIs pass.

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]

File: packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScanner.m [hunk 1]
```diff
@@ -314,12 +316,16 @@ + (NSString *)zxingFormatToString:(ZXBarcodeFormat)format
...
+#ifdef __IPHONE_15_4
       // available in iOS 15.4+
       if (@available(iOS 15.4, *)) {
         return AVMetadataObjectTypeCodabarCode;
       } else {
         return @"unknown";
       }
+#else
+      return @"unknown"
+#endif
```
Comment (tsapeta) [inline packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScanner.m:328]:
```suggestion
#ifdef __IPHONE_15_4
      if (@available(iOS 15.4, *)) {
        return AVMetadataObjectTypeCodabarCode;
      }
#endif
      return @"unknown"
```

File: packages/expo-barcode-scanner/ios/EXBarCodeScanner/Utilities/EXBarCodeScannerUtils.m [hunk 2]
```diff
@@ -142,12 +144,16 @@ + (NSString *)zxingFormatToString:(ZXBarcodeFormat)format
...
+#ifdef __IPHONE_15_4
       // available in iOS 15.4+
       if (@available(iOS 15.4, *)) {
         return AVMetadataObjectTypeCodabarCode;
       } else {
         return @"unknown";
       }
+#else
+      return @"unknown"
+#endif
```
Comment (tsapeta) [inline packages/expo-barcode-scanner/ios/EXBarCodeScanner/Utilities/EXBarCodeScannerUtils.m:156]:
```suggestion
#ifdef __IPHONE_15_4
      if (@available(iOS 15.4, *)) {
        return AVMetadataObjectTypeCodabarCode;
      }
#endif
      return @"unknown"
```

**Review by tsapeta — APPROVED** [review]
The switches can be simplified, suggestions below 👇

File: packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScanner.m (hunk 1 above)
Comment (barthap) [reply packages/expo-barcode-scanner/ios/EXBarCodeScanner/EXBarCodeScanner.m:328]: Missing semicolon
```suggestion
#ifdef __IPHONE_15_4
      if (@available(iOS 15.4, *)) {
        return AVMetadataObjectTypeCodabarCode;
      }
#endif
      return @"unknown";
```

**Comment by barthap** [comment]
Test suite failed due to network error 🤦‍♂️ but unit tests built successfuly so merging this now.

**Pull request merged**